### PR TITLE
Add event typing for `common` and `devp2p`

### PR DESCRIPTION
**Pull request by acolytec3**
This PR partially addresses #3750 

- Adds types for `devp2p` and `common` events
- Rewrites the `devp2p` `eth-simulator` test so it's comprehensible
- Removes the `les-simulator` tests

### CODE REVIEW

File: packages/devp2p/src/types.ts [hunk 1]
```diff
@@ -1,8 +1,66 @@
...
+import type { SNAP } from './protocol/snap.js'
+import type { Peer } from './rlpx/peer.js'
 import type { Common } from '@ethereumjs/common'
+import type { NestedUint8Array } from '@ethereumjs/rlp'
 import type { Socket } from 'net'
 
+export interface RLPxEvent {
+  'peer:added': Peer
+  'peer:error': { peer: Peer; error: any }
+  'peer:removed': { peer: Peer; reason: any; disconnectWe: any }
```
Comment (scorbajio) [inline packages/devp2p/src/types.ts:14]: What is `disconnectWe`? Is there a more descriptive name for this field?

File: packages/devp2p/test/integration/dpt-simulator.spec.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ describe('DPT simulator tests', () => {
   it('DPT: new working node', () => {
     const dpts = util.initTwoPeerDPTSetup(41622)
 
-    dpts[0].events.on('peer:new', async (peer: any) => {
```
Comment (scorbajio) [inline packages/devp2p/test/integration/dpt-simulator.spec.ts:11]: It's actually great to see these events typed in a more meaningful way 🙂, have struggled with what "peer" type is being returned when using the API before.

File: packages/devp2p/tsconfig.json [hunk 3]
```diff
@@ -5,10 +5,10 @@
     "test/**/*.ts",
     "examples/**/*.ts",
     "scripts/**/*.ts",
-    "examples/simple.cts",
```
Comment (scorbajio) [inline packages/devp2p/tsconfig.json:8]: Why did these change? Were these examples not being included before?

File: packages/devp2p/src/types.ts (hunk 1 above)
Comment (acolytec3) [reply packages/devp2p/src/types.ts:14]: It's just saying whether we "disconnected" the connection or the remote peer.  I'll add some comments on the types.

File: packages/devp2p/tsconfig.json (hunk 3 above)
Comment (acolytec3) [reply packages/devp2p/tsconfig.json:8]: We just forgot to update the extension when we migrated over to ESM for our base dev environment.  This is just miscellaneous cleanup.

File: packages/devp2p/src/types.ts [hunk 4]
```diff
@@ -1,8 +1,66 @@
...
+import type { SNAP } from './protocol/snap.js'
+import type { Peer } from './rlpx/peer.js'
 import type { Common } from '@ethereumjs/common'
+import type { NestedUint8Array } from '@ethereumjs/rlp'
 import type { Socket } from 'net'
 
+export interface RLPxEvent {
+  'peer:added': [peer: Peer]
+  'peer:error': [peer: Peer, error: any]
+  'peer:removed': [peer: Peer, reason: any, disconnectWe: any] // disconnectWe indicates whether the disconnection was initiated by us or not
```
Comment (gabrocheleau) [inline packages/devp2p/src/types.ts]: I typed the disconnectWe as `boolean | null`. It was typed as that in rlpx/peer.ts and typecheck passes

**Comment by gabrocheleau** [comment]
Pushed some minor type improvements and reviewed the code, lgtm!

**Review by gabrocheleau — APPROVED** [review]

**Pull request merged**